### PR TITLE
[editorial] Remove implementation detail from Logs Bridge API

### PR DESCRIPTION
**Pull request by MrAlias**
The bridge-api.md document specifies the Logs Bridge API, it does not defined the OpenTelemetry defined implementations of this API. Remove this detail from the API and leave it for the implementation specification to define.

Importantly, the API is decoupled from implementation for a reason. Third-party implementation may exists that OTel does not have domain over. This document needs to written with that in mind.

Related to https://github.com/open-telemetry/opentelemetry-specification/pull/3171 and https://github.com/open-telemetry/opentelemetry-specification/issues/3071 which were similar issues resolved for the metric API definition.

### CODE REVIEW

File: specification/logs/bridge-api.md [hunk 1]
```diff
@@ -46,9 +46,6 @@ graph TD
 
 `Logger`s can be accessed with a `LoggerProvider`.
 
-In implementations of the API, the `LoggerProvider` is expected to be the stateful
```
Comment (tigrannajaryan) [inline specification/logs/bridge-api.md:49]: This is wording copied from traces and metrics:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#tracerprovider
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#meterprovider

Is there a reason it makes sense there but not here? If not let's make sure we are consistent and remove from everywhere.

Comment (MrAlias) [reply specification/logs/bridge-api.md:49]: Agreed. There is more cleanup in those signals. I will try to get PRs together for them as well.

Comment (MrAlias) [reply specification/logs/bridge-api.md:49]: https://github.com/open-telemetry/opentelemetry-specification/pull/3890

Comment (MrAlias) [reply specification/logs/bridge-api.md:49]: Resolving this comment as that work is in progress and shouldn't block the successful merge of this PR.

**Review by pellared — APPROVED** [review]

**Review by trask — APPROVED** [review]

**Review by reyang — APPROVED** [review]

**Pull request merged**